src/api/db/migrate/20160317164117_sanitize_invalid_repository_names.rb

```diff
@@ -0,0 +1,17 @@
+class SanitizeInvalidRepositoryNames < ActiveRecord::Migration
+  def up
+    Repository.where("name REGEXP ?", "^_|[:\/]+").find_each do |repository|
+      repository.name = repository.name.gsub(/^_|[:\/]+/, "")
+      repository.name = "please_enter_a_valid_name_#{SecureRandom.hex(5)}" if repository.blank?
+      repository.save!
```

Comment:
this will not write it to the backend. so api and backend will be out of sync.

Problem is that backend is maybe not running during migration run, so it must be a delayed job.

However, I am not sure if it is worth the hazzle. The user may need anyway become aware of it and manual fix it.
Esp. when this happens on critical frozen projects.